cdk/visit/lambda_code/log_visit/log_visit.py

```diff
@@ -117,45 +119,45 @@ def registrationWorkflow(self, current_user):
         except ClientError as e:
             self.logger.error(e.response['Error']['Message'])
 
-    def addVisitEntry(self, current_user, location):
+    def addVisitEntry(self, current_user, location, tool):
+        
         # Get the current date at which the user logs in.
-        visit_date = datetime.datetime.now().timestamp()
-        # Convert visit_date to human-readable format
-        visit_date = datetime.datetime.fromtimestamp(
-            visit_date).strftime('%Y-%m-%d %H:%M:%S')
+        eastern_tz = dateutil.tz.gettz('US/Eastern')
+        visit_date = datetime.datetime.now(tz=eastern_tz)
```

Comment:
What exactly does the time zone do here? In [the docs](https://docs.python.org/3/library/datetime.html#datetime-objects) I see the following:

> classmethod datetime.now(tz=None)
> Return the current local date and time.
> 
> If optional argument tz is None or not specified, this is like [today()](https://docs.python.org/3/library/datetime.html#datetime.datetime.today), but, if possible, supplies more precision than can be gotten from going through a time.time() timestamp (for example, this may be possible on platforms supplying the C gettimeofday() function).
> 
> If tz is not None, it must be an instance of a tzinfo subclass, and the current date and time are converted to tz’s time zone.
> 
> This function is preferred over today() and [utcnow()](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow).

Does this mean that we’d be baking the UTC-5 (or UTC-4 depending on DST) offset into our Unix time stamps with this change? A Unix time stamp is the number of seconds that have elapsed since the Unix epoch, excluding [leap seconds](https://en.m.wikipedia.org/wiki/Leap_second). The Unix epoch is 00:00:00 [UTC](https://en.m.wikipedia.org/wiki/Coordinated_Universal_Time) on 1 January 1970. 

It doesn’t really make sense to bake the timezone into a Unix time stamp, except for at the moment that you’re displaying the data in a human readable format. Does this actually change the number of seconds we save to the ddb table?

Reply:
This is a good point. I didn't think about any of this. I definitely agree the normalized UTC-Z time is better. I switched to use the time library in python to handle this, as it required less code to accomplish the same thing. Also left the datetime object for the original table and reverted the timezone there, so all prior data in the old table will be the same as before.